### PR TITLE
updates to adherence alerts

### PR DESCRIPTION
**Pull request by conundrumgirl**
small refactor

### CODE REVIEW

File: src/components/adherence/summary/AdherenceAlerts.tsx [hunk 1]
```diff
@@ -1,39 +1,43 @@
 import {useAdherenceAlerts, useUpdateAdherenceAlerts} from '@components/studies/adherenceHooks'
-import Loader from '@components/widgets/Loader'
 import ConfirmationDialog from '@components/widgets/ConfirmationDialog'
+import Loader from '@components/widgets/Loader'
```
Comment (conundrumgirl) [inline src/components/adherence/summary/AdherenceAlerts.tsx:3]: ignore import changes. I just have a setting to sort them on save. In settings.json in vscode I have '"editor.codeActionsOnSave": {
        "source.organizeImports": true,
    },

File: src/components/adherence/summary/AdherenceAlerts.tsx [hunk 2]
```diff
@@ -1,39 +1,43 @@
...
+import {AdherenceAlert, AdherenceAlertCategory} from '@typedefs/types'
+import dayjs from 'dayjs'
+import React, {FunctionComponent, ReactElement} from 'react'
+import {BorderedTableCell, StyledLink} from '../../widgets/StyledComponents'
 
 type AdherenceAlertsProps = {
   studyId: string
 }
 
+type Action = 'READ' | 'UNREAD' | 'DELETE'
```
Comment (conundrumgirl) [inline src/components/adherence/summary/AdherenceAlerts.tsx:39]: make action a type

File: src/components/adherence/summary/AdherenceAlerts.tsx [hunk 3]
```diff
@@ -150,11 +154,9 @@ const AlertText: FunctionComponent<{alert: AdherenceAlert; studyId: string}> = (
 
 const AlertMenu: FunctionComponent<{
```
Comment (conundrumgirl) [inline src/components/adherence/summary/AdherenceAlerts.tsx:155]: make AlertMenu a 'dumb' component. it is clearer when the updates take place in the same place as data fetches so this component just tells it's parent what to do

File: src/components/adherence/summary/AdherenceAlerts.tsx [hunk 4]
```diff
@@ -274,12 +266,37 @@ const AdherenceAlerts: FunctionComponent<AdherenceAlertsProps> = ({studyId}) =>
...
+        onSuccess(data, variables, context) {
+          setUpdateError(undefined)
+        },
+        onError() {
+          setUpdateError(alertUpdateError as Error)
+        },
+        onSettled() {
+          setTimeout(() => {
+            setProcessingAlertId(undefined)
+          }, 500)
```
Comment (conundrumgirl) [inline src/components/adherence/summary/AdherenceAlerts.tsx:287]: the timeout allows for an update to happen before loader goes away

File: src/components/adherence/summary/AdherenceAlerts.tsx [hunk 5]
```diff
@@ -300,52 +317,52 @@ const AdherenceAlerts: FunctionComponent<AdherenceAlertsProps> = ({studyId}) =>
           </FormGroup>
         ))}
       </BoxForFilters>
-      <Table>
-        <TableBody>
-          {data !== undefined &&
-            data.items
-              /* .sort((a, b) => {
+      <Loader reqStatusLoading={isLoading}>
```
Comment (conundrumgirl) [inline src/components/adherence/summary/AdherenceAlerts.tsx:320]: moved the main loader to the body of the table

File: src/components/adherence/summary/AdherenceAlerts.tsx [hunk 6]
```diff
@@ -300,52 +317,52 @@ const AdherenceAlerts: FunctionComponent<AdherenceAlertsProps> = ({studyId}) =>
...
+                      $isDark={index % 2 === 1}
+                      sx={{
+                        padding: theme.spacing(0.5),
+                        borderLeft: 'none',
+                        borderTop: '1px solid #EAECEE',
+                        borderBottom: '1px solid #EAECEE',
+                      }}>
+                      <IconWithRoundBackground icon={ALERTS[alert.category].icon} isRead={alert.read} />
+                      <AlertText alert={alert} studyId={studyId} />
+                      <AlertMenu
```
Comment (conundrumgirl) [inline src/components/adherence/summary/AdherenceAlerts.tsx:344]: change to AlertMenu call

File: src/components/studies/adherenceHooks.ts [hunk 7]
```diff
@@ -38,10 +38,13 @@ export const ADHERENCE_KEYS = {
 
   detail: (studyId: string, userId: string) => [...ADHERENCE_KEYS.list(studyId), userId] as const,
 
-  alertsAll: (studyId: string) => 
-    [...ADHERENCE_KEYS.all, 'alerts', studyId],
-  alerts: (studyId: string, category: string, currentPage: number, pageSize: number) => 
-    [...ADHERENCE_KEYS.alertsAll(studyId), category, currentPage, pageSize]
+  alertsAll: (studyId: string) => [...ADHERENCE_KEYS.all, 'alerts', studyId],
```
Comment (conundrumgirl) [inline src/components/studies/adherenceHooks.ts:41]: this is just formatting difference

**Review by hallieswan — APPROVED** [review]

**Pull request merged**